src/main/java/com/artipie/asto/Key.java

```diff
@@ -18,7 +19,7 @@
  *
  * @since 0.6
  */
-public interface Key {
+public interface Key extends Comparable<Key> {
```

Comment:
I'd suggest add `Comparator<? super Key>` interface static field instead of extending `Comparable<Key>` - this way you don't need to implement `compareTo` in each implementation and avoid confusing behavior of default interface implementation. It's a better strategy for comparing different interface implementation; where implementing `Comparable` is a good way for comparing `final` classes.